litestar/datastructures/secret_values.py

```diff
@@ -0,0 +1,120 @@
+from typing import Any, Union
+
+
+class SecretValue:
+    """Represents a secret value that can be of type `str` or `bytes`."""
+
+    def __init__(self, secret_value: Union[str, bytes]):
+        """Initializes a SecretValue object with a secret value of type `str` or `bytes`.
+
+        Args:
+            secret_value (Union[str, bytes]): The secret value to be encapsulated.
+        """
+
+        self._secret_value = secret_value
+
+    def get_value(self) -> Union[str, bytes]:
+        """Returns the actual secret value.
+
+        Returns:
+            Union[str, bytes]: The secret value.
+        """
+
+        return self._secret_value
+
+    def _get_hidden_value(self) -> Union[str, bytes]:
+        """Return the hidden representation of the secret value.
+
+        Raises:
+            NotImplementedError: Always raised to enforce implementation in subclasses.
+        """
+
+        raise NotImplementedError("Subclasses must implement _get_hidden_value")
+
+    def __len__(self) -> int:
+        """Returns the length of the actual secret value.
+
+        Returns:
+            int: Length of the secret value.
+        """
+
+        return len(self.get_value())
+
+    def __str__(self) -> str:
+        """Returns a string representation of the hidden secret value.
+
+        Returns:
+            str: String representation of the hidden secret value.
+        """
+
+        return str(self._get_hidden_value())
+
+    def __repr__(self) -> str:
+        """Returns a string representation of the object for debugging purposes.
+
+        Returns:
+            str: String representation of the object.
+        """
+
+        class_name = self.__class__.__name__
+        return f"{class_name}({self._get_hidden_value()!r})"
+
+    def __eq__(self, other: Any) -> bool:
+        """Checks if the given object is equal to the current instance.
+
+        Args:
+            other (Any): The object to compare.
+
+        Returns:
+            bool: True if equal, False otherwise.
+        """
+
+        return isinstance(other, self.__class__) and self.get_value() == other._secret_value
```

Comment:
```suggestion
    def __eq__(self, other: object) -> bool:
        """Checks if the given object is equal to the current instance.

        Args:
            other: The object to compare.

        Returns:
            bool: True if equal, False otherwise.
        """

        return isinstance(other, self.__class__) and self.get_value() == other._secret_value
```

Reply:
Done